### ISSUE TITLE
URLs that lack locale code should redirect

### ISSUE THREAD

**Issue by peterbe** [mdn/yari#358]
Opening `https://developer.mozilla.org/docs/Web/HTML` 302 redirects to `https://developer.mozilla.org/en-US/docs/Web/HTML`. We should support this in Yari too.

**Comment by peterbe** [comment]
The client-side code can redirect from within some JS magic in the `react-router` integration but it will never be able to support returning 302. And in fact, if the S3 key doesn't exist, the S3 website will respond with 404 but then serve the `/index.html` which can do the react router stuff. But that will be too late. So most definitely this needs to be resolved in the CDN with our deployer's Lambda@Edge solution.

**Comment by peterbe** [comment]
A simple solution would be to add a one-liner to the Lambda@Edge that we need. 

```python
if request.url.startswith('/docs/'):
    return redirect(f'/en-US{request.url}')
```
That's brutally simple and once implemented it means we don't need to ever worry about it in the react-router code.

**Comment by peterbe** [comment]
This is the kind of thing we should do in Lambda@Edge in the "Origin request". It should be pretty easy to express with JS code.

**Comment by peterbe** [comment]
I guess the least friction would to be mimic what Kuma does. Right? 
Requests for `/docs/**` need to pick a locale and that is done based on: https://github.com/mdn/kuma/blob/3f826b64b937e09a93d37c5d0d9320a0f80079c0/kuma/core/i18n.py#L147-L207

This is the code to do the lookup for all and any URL. Not necessarily when the URL lacks a locale. But basically, it seems it does:

1. Look for the value to the cookie key `django_language` (wot?!)
2. Parse the `Accept-Language` from left to right and stop as soon as you've found one that passes a regex and is one of the ones we support. 
3. Fallback on `en-US`. 

Hmm. Consider that that `settings.LANGUAGE_COOKIE_NAME` comes from Django and not from Kuma (no mention of `LANGUAGE_COOKIE_NAME` in `settings/common.py`) And the fact that even if you change your language in the "Edit Profile" it doesn't set any cookie called `django_language`. So maybe it's part of your session cookie. Or, it's just not set at all. In fact, I have no idea how to set a cookie that would trigger there.

**Comment by peterbe** [comment]
Our list of valid locales comes from https://github.com/mdn/yari/blob/ace80567c577369538720003634a7d21108a39cb/content/constants.js#L34
One way or another, we need to get the same list into the Lambda@Edge function.

**Comment by escattone** [comment]
This is was fully addressed by #1425.
